### DISCUSSION TITLE
Is there an option for a condition on a DAG run?

### DISCUSSION THREAD

**Discussion by johanneszellinger** [apache/airflow#23257] [Q&A]
I'm thinking of the following use case:

- I have several DAGs which occasionally create temporary files. Sometimes, a DAG fails and doesn't clean up these files. 
- I would like to have an additional "Clean-up" DAG, that will simply delete remaining temp files from old DAGs. This might be a scheduled DAG to run once a day or once a week. 
- However, this additional DAG has to wait until no other DAG is running, otherwise I run the risk of deleting a needed file.
- Similarly, while the "Clean-up" DAG is running, all other DAGs will have to wait until it is finished.

The use case is just an example, but would something like this (let a scheduled DAG wait until no other DAGs are running) be possible?
I could think of some workaround with sensors maybe (using filesensor and a file as a lock or something similar?), but I was wondering if there is a better way to do this. 

Thanks for any input!

**Comment by zhuowang-linkedin** [comment]
I think you can use sensor to check something that indicates if clean-up is running. sensor can hold up the DAG execution until satisfied.
